core/src/main/java/org/hisp/dhis/android/core/trackedentity/internal/TrackedEntityAttributeReservedValueEndpointCallFactory.kt

```diff
@@ -72,8 +72,8 @@ internal class TrackedEntityAttributeReservedValueEndpointCallFactory internal c
         return TrackedEntityAttributeReservedValueCallProcessor(
             data.databaseAdapter,
             handler,
-            query.organisationUnit(),
-            query.trackedEntityAttributePattern(),
+            query.organisationUnit,
+            query.trackedEntityAttributePattern!!,
```

Comment:
This `!!` is risky. The parameter in the query could be nonNull. Or check the nullability in a previous method.

Maybe it was already like this before and this parameter is never null, but it still seems a potential issue.

Reply:
Downstream, the code is prepared to receive a nullable pattern, so I'll remove the not-null assertion operator and allow the pattern to be null there.